R/utils.R

```diff
@@ -344,25 +344,44 @@ do_call_handler <- function(name, fun, params = list(), verbose = 1) {
   handler <- function(condition = "Error") {
     function(c) {
       if (length(params) == 0) {
-        params_str <- ""
+        params_str <- " (params empty)."
       } else {
         params_str <- paste0(
           utils::capture.output(tibble::glimpse(params))[-1], collapse="\n"
         )
-        params_str <- paste0(" with params:\n", params_str)
+        params_str <- paste0(" with the following params:\n", params_str)
+      }
+      condition <- match.arg(
+        condition, choices = c("Error", "Warning", "Message")
+      )
+      msg_start <- if (condition == "Message") {
+        paste0("The message below")
+      } else {
+        paste0(c$message, "\nThe above ", tolower(condition))
+      }
+      msg <- paste0(
+        msg_start, " occurred while processing \"", name, "\"", params_str
+      )
+      metadata <- list(name = name, params = params, pid = Sys.getpid())
```

Comment:
Should we add `fun = fun` to this list too?

Reply:
`fun` in this context is `DGP$fit` or `Method$fit`, which is why I omitted it since that's probably not what the user wants and it would be a bit more complicated to get at the `*_fun`. Plus they can get that from their Experiment directly using the name. I guess it's an extra step, but if it feels too annoying as a user trying to debug things then maybe we can provide helpers to get at the data / funs.

Reply:
Oh I see. Yeah let's not add `fun` since its `DGP$fit` or `Method$fit`. I thought it was the `*_fun`. It's easy enough to get at the data / funs.